### PR TITLE
Feat/add rest and transitionstart events

### PR DESCRIPTION
**Pull request by looeee**
This PR adds the new `rest` event and corresponding `restEpsilon` parameter discussed in #206. 

Also added a new `rest-and-sleep` example to show the difference between the two events, and updated the readme. 

While making the example, I discovered there was no way to reliably disable the buttons when starting a new transition using one of the buttons (which call `.rotateTo` and `.dollyTo`) - so I added an additional event, `transitionstart`. This fires whenever any kind of transition is started, which can be either a user input or a method with `enableTransition = true`. 

I originally wanted `transitionstart` to fire only for a method with `enableTransition = true`, but _not_ for user control. However, internally user controls use methods such as `.rotate` so there's currently no way to differentiate these. Probably that's ok though. 

I checked in `dist/` by mistake, let me know if you want me to remove it to make the code easier to review.

### CODE REVIEW

**Comment by yomotsu** [comment]
Thanks for the PR! I'll take a look.

I should have indicated that it's better to ignore the dist folder...but it's okay.

File: src/CameraControls.ts [hunk 1]
```diff
@@ -111,6 +112,8 @@ export class CameraControls extends EventDispatcher {
 	verticalDragToForward = false;
 
 	boundaryFriction = 0.0;
+	
+	restEpsilon = 0.0025;
```
Comment (yomotsu) [inline src/CameraControls.ts:116]: Might be `restThreshold`...?

Comment (looeee) [reply src/CameraControls.ts]: OK, done. Also remove dist/

**Comment by yomotsu** [comment]
Thanks! Just merged!

**Pull request merged**